miden-lib/src/notes/mod.rs

```diff
@@ -2,11 +2,64 @@ use crate::memory::{
     CREATED_NOTE_ASSETS_OFFSET, CREATED_NOTE_CORE_DATA_SIZE, CREATED_NOTE_HASH_OFFSET,
     CREATED_NOTE_METADATA_OFFSET, CREATED_NOTE_RECIPIENT_OFFSET, CREATED_NOTE_VAULT_HASH_OFFSET,
 };
-use crypto::{hash::rpo::RpoDigest as Digest, StarkField, Word, WORD_SIZE};
+use assembly::ast::ProgramAst;
+use crypto::{hash::rpo::RpoDigest as Digest, StarkField, Word, WORD_SIZE, Felt};
 use miden_objects::{
-    notes::{NoteMetadata, NoteStub, NoteVault},
-    NoteError,
+    notes::{NoteMetadata, NoteStub, NoteVault, Note, NoteScript},
+    NoteError, accounts::AccountId, assets::Asset,
 };
+use crate::assembler::assembler;
+
+pub enum Script {
+    P2ID { target: AccountId },
+    P2IDR {
+        target: AccountId,
+        recall_height: u32,
+    },
+}
+
+pub fn create_note_with_script(
```

Comment:
No - this would be more generally available to provide a kind of "stronger typing" for standardized note scripts. One place where this can be used is in the client.